01_financial-well-being/housing-value.qmd

```diff
@@ -0,0 +1,904 @@
+    ) |>
+    mutate(
+      ratio_house_value_households = construct_ratio(share_house_value, share_households),
+      .by = c(race_ethnicity, ...)
+    ) |>
+    arrange(year, state, {{ geo_var }}, ...)
+}
+
+
+reshape_to_wide <- function(data, ...) {
+  data |>
+    pivot_wider(
+      id_cols = c(year, state, ...),
+      names_from = race_ethnicity,
+      values_from = c(ratio_house_value_households, ratio_house_value_households_quality),
+      names_glue = "{race_ethnicity}_{.value}"
+    ) %>%
+    rename_with(~ str_remove(.x, "ratio_") %>% paste0("ratio_", .), contains("ratio")) %>%
+    # order variables according to request
+    select(
+      year, state, ...,
+      ratio_black_nh_house_value_households, ratio_black_nh_house_value_households_quality,
+      ratio_hispanic_house_value_households, ratio_hispanic_house_value_households_quality,
+      ratio_other_nh_house_value_households, ratio_other_nh_house_value_households_quality,
+      ratio_white_nh_house_value_households, ratio_white_nh_house_value_households_quality
+    ) %>%
+    arrange(year, state, ...)
+}
+```
+
+```{r}
+#| label: calculate-metric
+
+ratio_house_value_households_county <- ratio_house_value_households(
+  share_households_county,
+  house_value_county
+) |>
+  reshape_to_wide(county)
+
+
+ratio_house_value_households_county_subgroup <- ratio_house_value_households(
+  share_households_county_subgroup,
+  house_value_county_subgroup,
+  "subgroup"
+) |>
+  reshape_to_wide(county, subgroup)
+
+ratio_house_value_households_place <-
+  ratio_house_value_households(
+    share_households_place,
+    house_value_place
+  ) |>
+  reshape_to_wide(place)
+
+ratio_house_value_households_place_subgroup <-
+  ratio_house_value_households(
+    share_households_place_subgroup,
+    house_value_place_subgroup,
+    "subgroup"
+  ) |>
+  reshape_to_wide(place, subgroup)
+```
+
+**Check:** Are there 3,143 counties and 486 places in the final output?
+
+Our metric county data has 3,143 counties as expected for all years prior to 
+2022. The metric places data also has the expected 486 places.
+
+```{r}
+#| label: check-number-of-counties-places
+
+stopifnot(
+  ratio_house_value_households_county |>
+    distinct(year, state, county) |>
+    count(year) |>
+    filter(year != 2022) |>
+    pull(n) == 3143
+)
+
+
+stopifnot(
+  ratio_house_value_households_place |>
+    distinct(year, state, place) |>
+    count(year) |>
+    filter(year == 2022) |>
+    pull(n) == 486
+)
+```
+
+
+However, for 2022, the number of counties is 3,146
+```{r}
+#| label: check-number-of-counties-2022
+#| error: true
+
+stopifnot(
+  ratio_house_value_households_county |>
+    distinct(year, state, county) |>
+    count(year) |>
+    filter(year == 2022) |>
+    pull(n) == 3143
+)
+```
+
+Further examination shows that there are 5 counties in our metric data that
+are not in the county populations file. 
+```{r}
+#| label: find-extra-counties
```

Comment:
@cdsolari how do you suggest we address this issue?

Reply:
Thanks! yes, in other metrics we are having these all be N/A for CT because the county conversion is such a mess and we found out about it so close to our deadline. It is just for metrics that have 2022 data, and it will only be for 2022 for now. Prior years should still have CT values and should be ok. This is something we will deal with in our next round. Thanks! @malcalakovalski & @kmartinchek